### DISCUSSION TITLE
Time Integrator Convergence study

### DISCUSSION THREAD

**Discussion by clmurphey** [su2code/SU2#1441] [Ideas]
I previously posted to the SU2 slack [here](https://su2devteam.slack.com/archives/C2A1JLGDR/p1636657760008000) asking about time integrators in SU2. I am interested in potentially running a temporal convergence study on at least one example found in the current [VandV repository](https://github.com/su2code/VandV) or the [tutorial repository](https://github.com/su2code/Tutorials). 

Unless I am mistaken, the [VandV repository](https://github.com/su2code/VandV) mostly focuses on spatial convergence (at least in the MMS examples I have looked at). Is this a correct assumption? Would you expect the error in the examples there to be dominated mostly by grid error? Would a time integrator convergence study even be useful for these problems and the SU2 community? Essentially, I anticipate looking at a few time integration methods - both explicit and implicit. I have noticed a few discussions around the implementation of [Explicit Euler](https://github.com/su2code/SU2/issues/1432), which sparked this idea. 

Since I am relatively new to SU2, I wanted to see if such a contribution would be valuable to this community. I would greatly appreciate any feedback you can provide. I am a first-year PhD student studying numerical methods and I would love to learn from this community. 

Thank you! 
-Corey

P.S., My apologies if I am not posting this in the correct forum/repository. I didn't know where to post.

**Comment by TobiKattmann** [comment]
Hi Corey,

first of all: welcome to the SU2-community :)

Concerning Usefulness: Any Contribution, Research is valuable. And on that journey you would probably find some bugs here and there + plus maybe introducing new ideas who to handle these code-bits altogether :) so never question that aspect. 

From searching through the Testcases `grep -ri TIME_MARCHING TestCases/ --include=*cfg` I think only the DG-FEM solver uses explicit TIME_STEPPING and the rest (the "normal" FVM-solvers) use Dual time stepping 2nd order (i.e. BDF2). So I guess just setting up with both Time integration schemes (+ BDF1 maybe) for a case would be interesting to see whether the same transient can be created.

You can also join the Developers meetings each Wednesday 4pm CET  (early for you 🌅 ) and talk a bit more about your plans and general setting. And those meetings are basically first come first solve about what to talk about anything related to working on/with SU2 👍 

I hope I got everything somewhat correctly, Tobi

**Reply by clmurphey** [reply]
Hi Tobi, 

Thank you so much for your response. My apologies for the delay in getting back to you. Behind the scenes, I have been working on starting this ["time convergence" study.](https://github.com/clmurphey/su2_time_conv) I started with two examples: laminar flat plate and Navier Stokes flow over a cylinder (though I think something might be off on my cylinder example). I had a few issues with periodic boundary conditions and explicit time discretization, which I think is related to [this issue](https://github.com/su2code/SU2/issues/1090). 

I have written up more details on this process [here](https://github.com/clmurphey/su2_time_conv/blob/main/README.md). The gist of it is, as expected, 2nd Order Dual Time Stepping converges faster. Explicit methods also converge faster, though I really didn't look at what the error or computation expense of these methods are. 

My disclaimer here is: there are likely some holes in my implementation here and vast room for improvement. I would appreciate any feedback you or others may be able to provide, especially on the configuration files. Thanks for building a great package. It has been fun to play around with the configuration files over the past month. 

-Corey

**Comment by bigfooted** [comment]
Hi Corey,

For the flow over a cylinder case that you performed, is it a 'standard' laminar unsteady vortex shedding flow? I saw in one of the cases that Re=40, this is below the critical Reynolds number where vortex shedding starts to occur in measurements (Re~50). Since we know the vortex shedding frequency (e.g. Williamson 1988, Hammache&Gharib,1991), it might be valuable to check if your setup reproduces the shedding frequency correctly for all your configurations. For low mach cases, you can also try using SOLVER= INC_NAVIER_STOKES.
